### PR TITLE
port2core :tada:

### PR DESCRIPTION
**Pull request by TerribleDev**
Port the main project to core. This still supports .net 3.5!!!

Convert unit tests to project.json format but maintain the same version of nunit/rinomocks (so obv the unit tests are still full framework bound).

also update appveyor accordingly, and set the proper package options in the project.json file since nuspec's dont really exist with project.json.

The one wildcard is I got rid of the Configuration project, and I'm not too sure what requires it, since I was able to compile without it.

if the configuration project is needed and has to be in the net35 nuget package just let me know.

### CODE REVIEW

**Comment by TerribleDev** [comment]
I'm assuming this makes #56  obsolete sorry :(

**Comment by TerribleDev** [comment]
btw I tried to make the changes minimal. We can add async overloads and what not later. One major change at a time!

**Comment by TerribleDev** [comment]
So I just realized the unit tests didn't compile because of the missing Configuration project. I see its 1 file that has an obsolete on it. Should we just kill it? For now I'll comment out the broken test file

**Comment by TerribleDev** [comment]
also this is a lingering question: https://github.com/Pereingo/statsd-csharp-client/pull/65/files#diff-73c83093cf1ddb7823ddb3f7572110b9R41

File: src/Tests/TCPSmokeTests.cs [hunk 1]
```diff
@@ -38,19 +38,20 @@ public void TearDownUdpListener()
 			// Stop listening for client requests.
 			tcpListener.Stop();
 		}
+        //this test requires a listener for a tcp socket...is it really a unit test?
```
Comment (DarrellMozingo) [inline src/Tests/TCPSmokeTests.cs]: Nope - it's more integration. To be fair it never did say it was a unit test ;)  We just lumped them all together.

File: src/Tests/NamingIntegrationTests.cs [hunk 2]
```diff
@@ -1,41 +1,41 @@
...
-{
-    [TestFixture]
-    public class NamingIntegrationTests
-    {
-        [Test]
-        public void environment_name()
-        {
-            Assert.That(Naming.CurrentEnvironment,Is.EqualTo("environment"));
-        }
+//namespace Tests
```
Comment (DarrellMozingo) [inline src/Tests/NamingIntegrationTests.cs]: Was actually planning to remove this in a future version as it's been Obsolete for a while - seems as good a time as any if you wouldn't mind just deleting the tests too.

**Comment by TerribleDev** [comment]
You place the nuspec info in the peoject.json file. I just realized I filled out the wrong file. So I will have to fix that, but it looks like this. https://github.com/Pereingo/statsd-csharp-client/pull/65/files#diff-30e4c60ec7fca7af35cd910663eb7df8R4

**Comment by TerribleDev** [comment]
You run `dotnet pack path/to/project.json` and that project gets placed in the nuget package.

**Comment by DarrellMozingo** [comment]
Ah, cool. So for in the `appveyor.yml` can we just do:
```
after_build:
  dotnet pack src\StatsdClient --output .
```
And we'll get the same artefact output, right?

**Comment by TerribleDev** [comment]
@DarrellMozingo isn't that what [I am already doing?](https://github.com/Pereingo/statsd-csharp-client/pull/65/files#diff-180360612c6b8c4ed830919bbb4dd459R14)

**Comment by DarrellMozingo** [comment]
Oh yea, I'm blind :P  Can you add the `--output .` or change the artefact path so they get picked up (looks like they default to the `bin` folder)?

**Comment by TerribleDev** [comment]
So I'm really sick with the flu right now. I'll have to come back to this next week when I'm feeling better.

**Comment by DarrellMozingo** [comment]
No worries mate, hope you feel better.

**Comment by dkhanaferov** [comment]
Will this make it out some time soon? We'd really like to be able to use this library in out new .net core apps.

**Comment by TerribleDev** [comment]
@dkhanaferov 
Oh sorry so I was super sick in november, and december..was well the holidays...and then forgot I had this on my plate

I'll make this happen very soon-ish.

**Comment by TerribleDev** [comment]
@DarrellMozingo ....done

**Comment by DarrellMozingo** [comment]
Thanks again @TerribleDev for the PR

**Pull request merged**